builder/main.py

```diff
@@ -109,6 +115,100 @@ def generate_uf2(target, source, env):
 if not env.get("PIOFRAMEWORK"):
     env.SConscript("frameworks/_bare.py")
 
+
+def convert_size_expression_to_int(expression):
+    conversion_factors = {
+        "M": 1024*1024,
+        "MB": 1024*1024,
+        "K": 1024,
+        "KB": 1024,
+        "B": 1,
+        "": 1 # giving no conversion factor is factor 1.
+    }
+    # match <floating pointer number><conversion factor>.
+    extract_regex = r'^((?:[0-9]*[.])?[0-9]+)([mkbMKB]*)$'
+    res = re.findall(extract_regex, expression)
+    # unparsable expression? Warning.
+    if len(res) == 0:
+        sys.stderr.write(
+            "Error: Could not parse filesystem size expression '%s'."
+            " Will treat as size = 0.\n" % str(expression))
+        return 0
+    # access first result
+    number, factor = res[0]
+    number = float(number)
+    number *= conversion_factors[factor.upper()]
+    return int(number)
+
+def fetch_fs_size(env):
+    # follow generation formulas from makeboards.py for Earle Philhower core
+    # given the total flash size, a user can specify
+    # the amount for the filesystem (0MB, 2MB, 4MB, 8MB, 16MB)
+    # via board_build.filesystem_size,
+    # and we will calculate the flash size and eeprom size from that.
+    flash_size = board.get("upload.maximum_size")
+    filesystem_size = board.get("build.filesystem_size", "0MB")
+    filesystem_size_int = convert_size_expression_to_int(filesystem_size)
+
+    maximum_sketch_size = flash_size - 4096 - filesystem_size_int
```

Comment:
`4096` is used all over the function. Is it `FS_BLOCK` or something similar? Any chance to give `4096` a proper name? Or, at least, mention this value in comments?

Reply:
A variable and comment was added to make this clearer.